pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -296,38 +419,52 @@ func flavorSelector(spec *corev1.PodSpec, allowedKeys sets.String) nodeaffinity.
 	return nodeaffinity.GetRequiredNodeAffinity(&corev1.Pod{Spec: specCopy})
 }
 
-// fitsFlavorLimits returns whether a requested resource fits in a specific flavor's quota limits.
+// fitsFlavorLimits returns how could this flavor be assigned, according to the
+// remaining quota in the ClusterQueue and cohort.
 // If it fits, also returns any borrowing required.
-func fitsFlavorLimits(rName corev1.ResourceName, val int64, cq *cache.ClusterQueue, flavor *cache.FlavorLimits) (int64, *Status) {
+func fitsFlavorLimits(rName corev1.ResourceName, val int64, cq *cache.ClusterQueue, flavor *cache.FlavorLimits) (FlavorAssignmentMode, int64, *Status) {
 	var status Status
 	used := cq.UsedResources[rName][flavor.Name]
+	mode := NoFit
+	if val <= flavor.Min {
+		// The request can be satisfied by the min quota, if all active workloads
+		// in the ClusterQueue are preempted.
+		mode = ClusterQueuePreempt
+	}
+	borrow := used + val - flavor.Min
+	if borrow <= 0 {
+		borrow = 0
+		// The request can be satisfied by the min quota, if all active workloads
+		// from other ClusterQueues in the cohort are preempted.
+		mode = CohortReclaim
+	}
 	if flavor.Max != nil && used+val > *flavor.Max {
 		status.append(fmt.Sprintf("borrowing limit for %s flavor %s exceeded", rName, flavor.Name))
-		return 0, &status
+		return mode, 0, &status
```

Comment:
this if statement will never be entered if we entered the previous one, perhaps hook them via an else to emphasize that we are now looking at whether we can borrow resources:

`} else if flavor.Max != nil && used+val > *flavor.Max {`